packages/sdk-btc/source/client.service.ts

```diff
@@ -13,17 +15,36 @@ export class ClientService extends Services.AbstractClientService {
 	public override async transactions(
 		query: Services.ClientTransactionsInput,
 	): Promise<Collections.ConfirmedTransactionDataCollection> {
-		if (!query.addresses) {
-			throw new Error("No addresses specified for querying for transactions");
+		if (!query.addresses && !query.senderPublicKey) {
+			throw new Error("Need specify either addresses or a extended public key for querying for transactions");
 		}
 
-		const response = await this.#post("wallets/transactions", { addresses: query.addresses });
+		let addresses = query.addresses;
+		if (!addresses) {
+			const network = getNetworkConfig(this.configRepository);
+
+			const bip = bip44; // TODO Make this a method param or part of the ClientTransactionsInput
+
+			const xpub = query.senderPublicKey!;
+			addresses = (await this.usedAddresses(addressGenerator(bip, network, xpub, true, 100))).concat(
+				await this.usedAddresses(addressGenerator(bip, network, xpub, false, 100)),
+			);
+		}
+
+		const response = await this.#post("wallets/transactions", { addresses });
 
 		return this.dataTransferObjectService.transactions(response.data, this.#createMetaPagination(response));
 	}
 
-	public override async wallet(id: string): Promise<Contracts.WalletData> {
-		const response = await this.#post(`wallets`, { addresses: [id] });
+	public override async wallet(xpub: string): Promise<Contracts.WalletData> {
+		const network = getNetworkConfig(this.configRepository);
+
+		const bip = bip44; // TODO Make this a method param
```

Comment:
And here... we're assuming a wallet is simply an id. In the case of btc, and only if it is an xpub) we also need to know what addressing scheme to use. xpub will be the same for all 3 methods, but each will yield different addresses. So it *is* part of the wallet "identity" or metadata. And this method `#wallet()` only takes an id, so we'll need to change it.

I'm at a point where I could achieve what we need in terms of data on the bitcoin side, but I'm a bit lost within the SDK. [Here](https://github.com/PayvoHQ/profiles/blob/master/source/transaction-index.ts#L51) is the only place where we call this method from profiles, and given we have a `IReadWriteWallet` I think it should be perfectly possible to access what bip to use. But, even if available (when we have it available) we have no means to pass it. So we'll need new signatures for contract methods.

Can you help?